services/authoring/src/repositories/git-repository.js

```diff
@@ -64,23 +66,31 @@ class GitRepository {
     return entry.isBlob() ? (await entry.getBlob()).toString() : undefined;
   }
 
-  async getHistory(fileNames, { revision, maxCount = 1000 } = {}) {
+  async getHistory(fileNames, { revision } = {}) {
     fileNames = Array.isArray(fileNames) ? fileNames : [fileNames];
 
-    const sha = revision || (await this._repo.getMasterCommit()).sha();
-    const walker = this._repo.createRevWalk();
-    walker.sorting(git.Revwalk.SORT.TIME);
-
-    const historyEntries = await mapSeries(fileNames, (fileName) => {
-      walker.push(sha);
-      return walker.fileHistoryWalk(fileName, maxCount);
-    });
-
-    const mapEntry = ({ commit }) => ({
-      sha: commit.sha(),
-      author: commit.author().name(),
-      date: commit.date(),
-      message: commit.message(),
+    const historyEntries = await Promise.all(fileNames.map(async (file) => {
+      console.time(file);
+      const options = [
+        '--since="1 month ago"',
```

Comment:
I think we should either remove since entirely, or make it configurable like history_count